### PR TITLE
chore: port to python 3

### PR DESCRIPTION
**Pull request by ivan-aksamentov**
Ports the project to Python 3

See https://github.com/neherlab/ffpopsim/blob/python3/doc/docker.md

TL;DR:

```bash
./docker-dev make -j $(nproc)
./docker-dev ./run-examples
```

Items:
 - [x] Port to Python 3 (Works with 3.6 - 3.11)
 - [x] Port to Swig 4
 - [x] Add continuous integration (https://github.com/neherlab/ffpopsim/actions). You can download prebuilt wheels in the dist.zip file in the "Artifacts" section of a build (e.g. https://github.com/neherlab/ffpopsim/actions/runs/3478514021) and then unzip and install an appropriate wheel with pip. For example for Linux and Python 3.8 it will be:
   ```bash
   pip3 install FFPopSim-2.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
   ```
 - [x] Add release routine:
   ```bash
   ./scripts/release 3.0.0
   ```
 - [x] Add publishing routine: every git tag (created in the release routing) triggers CI build and publication to PyPi, so that the package can be installed with 
   ```bash
   pip3 install FFPopSim
   ```

Limitations and further work:

 - Windows is not supported (only Linux and macOS)

### CODE REVIEW

**Comment by ivan-aksamentov** [comment]
@rneher @anna-parker @PierreBarrat I made it such that the project builds with Python 3.6 to 3.11 and with more recent packages, and that all Python examples work.

There are many warnings, both during build and on runtime, some of them quite serious.

In order for this to be useful, someone need to check that this version does reasonable things and that it outputs expected outputs. Would be nice if you could check it and compare with the known good outputs from Python 2 version.

I have no scientific background or any understanding of the project, so this task will be almost impossible for me to achieve in reasonable time.

If you find that Python 3 version works sufficiently well, we can merge it to master and release as a PyPi package (I've also setup the CI and the release routine).

Let me know if you notice anything strange.

**Comment by rneher** [comment]
I tested this and couldn't spot any problems. Where we have clear theory expectations, the results match.

**Review by rneher — APPROVED** [review]

**Comment by iosonofabio** [comment]
Let me know if you want me to go through the warnings at some point - I could find she time next week or so

**Comment by ivan-aksamentov** [comment]
@iosonofabio Thanks Fabio! If you find a spare minute, then totally do. The `make` targets are roughly the same. Or you can use the docker setup as described in the first message.

**Comment by ivan-aksamentov** [comment]
I am going to merge this and release as 3.0.0
